CMakeLists.txt

```diff
@@ -187,31 +187,36 @@ if(WITH_BLUESTORE)
 endif()
 
 include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(WITH_LIBBLK "Enable backend driver to access storage device" ON
+  "WITH_BLUESTORE OR WITH_RBD_RWL" OFF)
+
 CMAKE_DEPENDENT_OPTION(WITH_LIBURING "Enable io_uring bluestore backend" OFF
-  "WITH_BLUESTORE;HAVE_LIBAIO" OFF)
+  "WITH_LIBBLK;HAVE_LIBAIO" OFF)
 set(HAVE_LIBURING ${WITH_LIBURING})
 
 CMAKE_DEPENDENT_OPTION(WITH_SYSTEM_LIBURING "Require and build with system liburing" OFF
-  "HAVE_LIBAIO;WITH_BLUESTORE" OFF)
+  "WITH_LIBBLK;HAVE_LIBAIO" OFF)
 
-CMAKE_DEPENDENT_OPTION(WITH_BLUESTORE_PMEM "Enable PMDK libraries" OFF
-  "WITH_BLUESTORE" OFF)
+CMAKE_DEPENDENT_OPTION(WITH_PMEM "Enable PMDK libraries" OFF
+  "WITH_LIBBLK" OFF)
+if(WITH_PMEM)
+  if(NOT WITH_LIBBLK)
+    message(SEND_ERROR "Please enable LIBBLK for using PMDK")
+  endif()
+  set(HAVE_PMEM ON)
+endif(WITH_PMEM)
 
 CMAKE_DEPENDENT_OPTION(WITH_RBD_RWL "Enable librbd persistent write back cache" OFF
   "WITH_RBD" OFF)
 
 CMAKE_DEPENDENT_OPTION(WITH_SYSTEM_PMDK "Require and build with system PMDK" OFF
-  "WITH_RBD_RWL OR WITH_BLUESTORE_PMEM" OFF)
-
-if(WITH_BLUESTORE_PMEM)
-  set(HAVE_BLUESTORE_PMEM ON)
-endif()
+  "WITH_RBD_RWL OR WITH_PMEM" OFF)
 
 CMAKE_DEPENDENT_OPTION(WITH_SPDK "Enable SPDK" OFF
   "CMAKE_SYSTEM_PROCESSOR MATCHES i386|i686|amd64|x86_64|AMD64|aarch64" OFF)
 if(WITH_SPDK)
-  if(NOT WITH_BLUESTORE)
-    message(SEND_ERROR "Please enable WITH_BLUESTORE for using SPDK")
+  if(NOT WITH_LIBBLK)
```

Comment:
might want to update the line defining `WITH_SPDK` option like
```cmake
CMAKE_DEPENDENT_OPTION(WITH_SPDK
  "Enable SPDK" OFF
  "WITH_LIBBLK;CMAKE_SYSTEM_PROCESSOR MATCHES i386|i686|amd64|x86_64|AMD64|aarch64" OFF)
```

Reply:
good suggestion

Reply:
i need to understand what WITH_LIBBLK is for first. the suggestion is solely made based on the code itself.

Reply:
@tchaikov 
Originally, SPDK could be only used by BLUESTORE. NOW, SPDK is going to be integrated into libblk. Other components could also SPDK through the libblk library.